docs: reference for the fetch-service integration (#903)
canonical/craft-application: docs/common/craft-application/reference/fetch-service.rst

necessary during the build itself, like compilers and other development packages.

There are two modes of integration between |Starcraft| builds and Fetch Service
sessions: managed, and external.

Managed sessions
----------------

When the ``--enable-fetch-service`` option is passed to the ``pack`` command,
|Starcraft| will take care of creating a Fetch Service session before starting the
build and shutting it down afterwards. In this scenario the Fetch Service needs to be
available on the local host so that |Starcraft| can talk to it.

The session created by |Starcraft| can be either "permissive", in which case the Fetch
Service will inspect the traffic but not block accesses that would be otherwise
restricted, or "strict", in which case only traffic that is permitted under the Fetch
Service's own configuration will be allowed. This constitutes the session's *policy*,
and can be selected by passing either ``permissive`` or ``strict`` to
``--enable-fetch-service=<policy>``.

Since the application controls the session, it also creates the final session report
with the description of the items that were downloaded during the packing of the

> artifact. This is provided as a JSON file with the same name as the artifact itself.

Author: oh sorry I did update this:
<img width="817" height="91" alt="image" src="https://github.com/user-attachments/assets/2f197c97-ffa1-4b6c-896b-84a0f6f40ff4" />

Fetch Service sessions
======================

.. admonition:: Experimental
    :class: important

    The integration with the Fetch Service is an experimental feature and therefore
    subject to change.

|app| uses Fetch Service to validate all dependencies that are downloaded
during |artifact-indefinite| build. These dependencies are either software
included in the |artifact| itself, or software that the build tool or build system
needs.

> Fetch Service operates in *sessions*, which can be employed in one of two modes:

Reviewer: ```suggestion
Fetch Service operates in *sessions*, which run in one of two modes:
```

Nitpick.

description of the items that were downloaded during artifact build. This is provided
as a JSON file with the same name as the artifact itself, in the same directory.

External sessions
-----------------

Alternatively, |app| can be configured to use a pre-existing Fetch Service
session in its builds. In this mode of operation, |app| configures all
traffic filtering for the session, but you are responsible for opening and closing
the session itself, and otherwise managing Fetch Service.

In order to make use of this mode, users must create a Fetch Service session and
configure the following environment variables before invoking the ``pack`` command:

.. list-table::
    :header-rows: 1

    * - Variable
      - Description
    * - ``http_proxy``, ``https_proxy``
      - Must point to the full session url, including server and port.
    * - ``CRAFT_PROXY_CERT``
      - Must point to the Fetch Service's CA certificate. This file must be locally
        accessible by |app|.
    * - ``CRAFT_USE_EXTERNAL_FETCH_SERVICE``
      - Must be ``1``.

Because |app| has no control over the Fetch Service session in this mode, it

> can't create a session report.

Author: unfortunately all documentation we have for that is private at the moment
